src/Sulu/Bundle/AdminBundle/Resources/js/components/containerSizeAware/containerSizeAware.js

```diff
@@ -0,0 +1,59 @@
+// @flow
+import {action, observable} from 'mobx';
+import React from 'react';
+import {observer} from 'mobx-react';
+import styles from './containerSizeAware.scss';
+
+export default function containerSizeAware(Component: ReactClass<*>, containerClass: string = styles.container) {
+    @observer
+    class ContainerSizeAwareComponent extends React.Component {
+        component: *;
+        container: HTMLElement;
+        @observable containerWidth: number = 0;
+        @observable containerHeight: number = 0;
+
+        componentDidMount() {
+            window.addEventListener('resize', this.handleWindowResize);
+            if (this.component.containerDidMount) {
+                window.requestAnimationFrame(this.component.containerDidMount);
```

Comment:
Not this component has to guarantee, that the passed children are loaded. The class who renderes the `RectangleSelection` (which is wrapped with this HOC) has to guarantee, that on rendering everything passed to the `RectangleSelection` is already fully loaded.